src/app.js

```diff
@@ -97,25 +98,21 @@ export default function (app) {
 			subscriptions = subscriptions.concat(app.subscriptions)
 		}
 
-		if (app.hooks) {
-			hooks.push(app.hooks)
-		}
-	}
-
-	function hook(name, fn) {
-		for (var key in hooks) {
-			var hookfn = hooks[key][name]
-			if (hookfn) {
-				fn(hookfn)
-			}
+		var _hooks = app.hooks
+		if (_hooks) {
+			Object.keys(_hooks).forEach(function (key) {
+				if (!hooks[key]) {
+					hooks[key] = []
+				}
+				hooks[key].push(_hooks[key])
+			})
 		}
-		return hookfn !== undefined
 	}
-
+	
 	function render(model, view) {
-		hook("onRender", function (fn) {
-			view = fn(model, view)
-		})
+		for (var i in hooks.onRender) {
```

Comment:
@dodekeract Using for in here is the easiest way to avoid checking if onRender is undefined.